pkg/apis/migration/v1alpha1/gvr.go

```diff
@@ -0,0 +1,32 @@
+package v1alpha1
+
+import "k8s.io/apimachinery/pkg/runtime/schema"
+
+// Incompatible - list of namespaces containing incompatible resources for migration
+// which are being selected in the MigPlan
+type Incompatible struct {
+	Namespaces []IncompatibleNamespace `json:"incompatibleNamespaces,omitempty"`
+}
+
+// IncompatibleNamespace - namespace, which is noticed
+// to contain resources incompatible by the migration
+type IncompatibleNamespace struct {
+	Name string            `json:"name"`
+	GVRs []IncompatibleGVR `json:"gvrs"`
```

Comment:
Can you help me understand why this is GVR instead of GVK?  The field seems to contain kinds (Eg: pods).  Since resources are instances of kinds, I would expect a field named GVR to identify instances by name.

Reply:
For the most part it is explained by the dynamic client API, on which the feature was built on. It expects a `GVR` structure to fetch the resource, and the actual Kind was lost during transformations. It is not suitable, as the resources are lowercase and in plural tense, which allows it to be directly translated to the API path. You could fetch `apis/extensions/v1beta1/replicasets`, but server does not recognize `apis/extensions/v1beta1/ReplicaSet` as a valid path.